osrdyne/src/drivers/docker.rs

```diff
@@ -0,0 +1,196 @@
+use std::{collections::HashMap, future::Future, pin::Pin};
+
+use bollard::{
+    container::{
+        Config, CreateContainerOptions, NetworkingConfig, RemoveContainerOptions,
+        StartContainerOptions,
+    },
+    secret::HostConfig,
+    Docker,
+};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::Key;
+
+use super::{
+    worker_driver::{DriverError, WorkerDriver, WorkerMetadata},
+    LABEL_CORE_ID, LABEL_INFRA_ID, LABEL_MANAGED_BY, MANAGED_BY_VALUE,
+};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct DockerDriverOptions {
+    /// The name of the Docker image to use for the core
+    pub core_image: String,
+    /// The prefix to use for the container names
+    pub container_prefix: String,
+    /// The command to run in the container
+    pub container_command: Option<Vec<String>>,
+    /// The default environment variables to set for the core
+    pub default_env: Vec<String>,
+    // Should the containers be started in host networking mode
+    pub host_networking: Option<bool>,
+}
+
+pub struct DockerDriver {
+    client: Docker,
+    options: DockerDriverOptions,
+    amqp_uri: String,
+}
+
+impl DockerDriver {
+    pub fn new(options: DockerDriverOptions, amqp_uri: String) -> DockerDriver {
+        DockerDriver {
+            client: Docker::connect_with_socket_defaults().expect("Failed to connect to Docker"),
+            options,
+            amqp_uri,
+        }
+    }
+}
+
+impl WorkerDriver for DockerDriver {
+    fn get_or_create_core_pool(
+        &self,
+        infra_id: Key,
+    ) -> Pin<Box<dyn Future<Output = Result<Uuid, DriverError>> + Send + '_>> {
```

Comment:
I got the impression from reading the doc that there was a desire to make it core agnostic.

```suggestion
    fn get_or_create_worker_pool(
        &self,
        worker_key: Key,
    ) -> Pin<Box<dyn Future<Output = Result<Uuid, DriverError>> + Send + '_>> {
```